product_abc_classification/models/abc_classification_profile.py

```diff
@@ -172,23 +193,43 @@ def _get_sort_key_percentage(rec):
                 totals["value"] += product_data["value"]
             product_list.sort(reverse=True, key=_get_sort_key_value)
             levels = profile.level_ids.sorted(
-                key=_get_sort_key_percentage, reverse=True
+                key=method_dict[profile.classification_type], reverse=True
             )
-            level_percentage = [[level, level.percentage] for level in levels]
-            current_value = 0
-            accumulated_percentage = level_percentage[0][1]
-            for product_data in product_list:
-                # Accumulated current value
-                current_value += product_data["value"] or 0.0
-                # This comparison would be the same as:
-                # current_value * 100 / totals["value"] > accumulated_percentage,
-                # but it is written in the next way to avoid division and decimal lost.
-                while (
-                    current_value * 100 > accumulated_percentage * totals["value"]
-                    and len(level_percentage) > 1
-                ):
-                    level_percentage.pop(0)
-                    accumulated_percentage += level_percentage[0][1]
-                product_data["product"].abc_classification_level_id = level_percentage[
-                    0
-                ][0]
+            if profile.classification_type == "percentage":
+                level_percentage = [[level, level.percentage] for level in levels]
+                current_value = 0
+                accumulated_percentage = level_percentage[0][1]
+                for product_data in product_list:
+                    # Accumulated current value
+                    current_value += product_data["value"] or 0.0
+                    # This comparison would be the same as:
+                    # current_value * 100 / totals["value"] > accumulated_percentage,
+                    # but it is written in the next way to avoid division and decimal lost.
+                    while (
+                        current_value * 100 > accumulated_percentage * totals["value"]
+                        and len(level_percentage) > 1
+                    ):
+                        level_percentage.pop(0)
+                        accumulated_percentage += level_percentage[0][1]
+                    product_data[
+                        "product"
+                    ].abc_classification_level_id = level_percentage[0][0]
+            elif profile.classification_type == "fixed":
+                if product_list:
+                    zero_level = profile.level_ids.filtered(
+                        lambda l: float_is_zero(l.fixed, precision_digits=2)
+                    )
+                    self.env["product.product"].search(
+                        [("abc_classification_profile_id", "=", profile.id)]
+                    ).write({"abc_classification_level_id": zero_level.id})
```

Comment:
```suggestion
                    products = self.env["product.product"]
                    for product_data in product_list:
                        products |= product_data["product"]
                    products.write({"abc_classification_level_id": zero_level.id})
```
no need of doing a search here.

Also, I don't understand why you add this write. Later you are doing another write in the following loop.

Reply:
What we are searching with this write is reset the value of all products with this profile to the minimum value of the levels. 

And the search is needed because we want to reset all, not just the products between the period that we have defined

Reply:
ok